client/asset/eth/polygon_bridge.go

```diff
@@ -135,6 +143,34 @@ func (b *polygonBridgePolygonERC20) getCompletionData(ctx context.Context, bridg
 	return common.Hex2Bytes(strings.TrimPrefix(res.Result, "0x")), nil
 }
 
+func (b *polygonBridgePolygonERC20) requiresCompletion() bool {
+	return false
+}
+
+// verifyPolygonBridgeCompletion is called for bridges from ethereum to polygon
+// POS, which do not require a completion transaction. It checks if the funds
+// have been allocated on polygon POS. The initiation transaction contained a log
+// with a state ID, and we check the LastStateId on the state receiver on polygon POS to
+// see if it is equal or greater than the required state ID.
+func verifyPolygonBridgeCompletion(ctx context.Context, cb bind.ContractBackend, stateReceiver *polygonbridge.StateReceiver, completionData []byte) (bool, error) {
```

Comment:
I think for base we need to use the other chain so the other wallet to verify the bridge completed. Any idea how to handle that?

Reply:
Base should have a separate function to verify that the bridge completed because it requires different logic. I think for base you have the check the successfulMessage map on this contract `https://basescan.org/address/0x4200000000000000000000000000000000000007`.

https://grok.com/share/bGVnYWN5_7c6fffdb-e208-4b59-90ec-168c854458fc

Reply:
If from eth that would be the to chain.